added double lepton DQM

clint-richardson opened the pull request:
adding new dqm for proposed dilepton triggers

clint-richardson committed:
added double lepton DQM

cmsbuild commented:
A new Pull Request was created by  @clint-richardson (Clint Richardson) for CMSSW_8_1_X.

It involves the following packages:

HLTriggerOffline/B2G


@cmsbuild, @vanbesien, @deguio, @davidlange6 can you please review it and eventually sign? Thanks.
@jdolen, @rappoccio this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 4e4d0f0b3801f59972cd8082bc49699d95187030
 When I ran the RelVals I found an error in the following worklfows: 
 5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>
1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>
25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>
10021.0 step3
<pre>runTheMatrix-results/10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017/step3_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017.log

</pre>
50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>
10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017.log

</pre>
25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12190/summary.html

clint-richardson committed:
fixed typo

cmsbuild commented:
Pull request #13930 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

clint-richardson commented:
Hello - sorry for the typo! I've fixed this and updated the pull request. If there is a way for me to run the tests on my own to save anymore potential back and forth please let me know.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 7757b210d5203402fc734f1ba1dba8486f6b42a3
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12196/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12196/summary.html

clint-richardson committed:
updated names for TSG request

cmsbuild commented:
Pull request #13930 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

clint-richardson commented:
updated to reflect request from TSG about path naming convention

clint-richardson commented:
hello, could you please retest these?

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2a6de14fca6eb7df0a5502be504d1df5781bd398
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12465/summary.html

deguio commented on HLTriggerOffline/B2G/python/b2gHLTValidationHarvest_cff.py:
```diff
@@ -13,6 +13,26 @@
         resolution = cms.vstring(""),
         )
 
+b2gDoubleLeptonEleMuHLTValidationHarvest = cms.EDAnalyzer("DQMGenericClient",
```
is this harvester added to any central harvesting sequence?
I don't see it in the b2gHLTriggerValidationHarvest one.

deguio commented on HLTriggerOffline/B2G/python/b2gHLTValidationHarvest_cff.py:
```diff
@@ -13,6 +13,26 @@
+        resolution = cms.vstring(""),
+        )
+
+b2gDoubleElectronHLTValidationHarvest = cms.EDAnalyzer("DQMGenericClient",
```
same here

deguio commented:
see comment above. apart from it this can go in.
sorry for being late,
F.

clint-richardson committed:
added to b2gharvesting

cmsbuild commented:
Pull request #13930 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

clint-richardson commented:
Hi @deguio  - does this update satisfy? Out of curiosity, even without this update when I run the harvesting step the DQM root files show up with the correct histograms...does this mean it must have been somewhere else anyway?

clint-richardson commented:
please test

clint-richardson commented:
..or I guess can someone else say the magic phrase...?

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12465/summary.html

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented:
ciao clint,
also the Efficiency plots show up? that would be weird since they are produced in the harvesting...
referring for example to: hEffLeptonEta
F.

cmsbuild commented:
+1
Tested at: dc2cf1d9e007c94b00234310dbea7b9f7ef6dea1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12493/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12493/summary.html

clint-richardson commented:
Hi @deguio , can you let me know if there is any other issue preventing the sign-off?

deguio commented:
+1

deguio commented:
please test

cmsbuild commented:
+1
Tested at: dc2cf1d9e007c94b00234310dbea7b9f7ef6dea1
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12523/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12523/summary.html

There are some workflows for which there are errors in the baseline:
 134.911 step 3
140.53 step 2
1000.0 step 2
1001.0 step 2
1003.0 step 2
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

davidlange6 commented on HLTriggerOffline/B2G/src/B2GDoubleLeptonHLTValidation.cc:
```diff
@@ -0,0 +1,167 @@
+
+    TString name = triggerNames.triggerNames()[i].c_str();
+    for (unsigned int j=0; j<vsPaths_.size(); j++) {
+      if (name.Contains(TString(vsPaths_[j]), TString::kIgnoreCase)) {
```
@clint-richardson - why should this comparison be case insensitive?

clint-richardson commented:
Hi @davidlange6 - I copied this code from existing B2G DQM modules (e.g. https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/HLTriggerOffline/B2G/src/B2GSingleLeptonHLTValidation.cc#L116). Unfortunately, I didn't have a part in writing them from scratch, and am not sure off the top of my head the motivation for it. Perhaps @rappoccio knows the answer to your question?

Also, just for my understanding, is it an issue to have it be case insensitive? Performance wise it should not affect the modules because HLT paths never differ only by capitalization.

davidlange6 commented:
> On Apr 22, 2016, at 9:57 AM, Clint Richardson <notifications@github.com> wrote:
> 
> Hi @davidlange6 - I copied this code from existing B2G DQM modules (e.g. https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/HLTriggerOffline/B2G/src/B2GSingleLeptonHLTValidation.cc#L116). Unfortunately, I didn't have a part in writing them from scratch, and am not sure off the top of my head the motivation for it. Perhaps @rappoccio knows the answer to your question?
> 
> Also, just for my understanding, is it an issue to have it be case insensitive? Performance wise it should not affect the modules because HLT paths never differ only by capitalization.
> 

the case insensitivity appears to imply two conversions from std::string to TString for each comparison done (for code simplicity reasons, it of course could be done with std::string directly). It would be best to avoid that completely.

> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>

clint-richardson commented:
@rappoccio do you have any thoughts on this change?

rappoccio commented:
Sorry, I missed this in the thread. I'm fine with making this case sensitive, at one point this was copied from another DQM module itself so that was not really a decision.

clint-richardson committed:
Update B2GDoubleLeptonHLTValidation.cc

removed case insensitive option for TString::Contains query

cmsbuild commented:
Pull request #13930 was updated. @cmsbuild, @dmitrijus, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: b463f658b11835cd1ee2871388b35872982af541
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12897/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/12897/summary.html

davidlange6 commented:
@clint-richardson 
so the last change is to avoid TString

if (triggerNames.triggerNames()[i].find(vsPaths_[j]) != std::string::npos) { //fill the histogram

clint-richardson committed:
removed use of TString

cmsbuild commented:
Pull request #13930 was updated. @cmsbuild, @dmitrijus, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

clint-richardson committed:
quick fix

cmsbuild commented:
Pull request #13930 was updated. @cmsbuild, @dmitrijus, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: a6d8cfa3f158334ddceda4a8785800c94d50fc37
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/13049/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13930/13049/summary.html

clint-richardson commented:
Hi @davidlange6 - could you let me know if there is anything else?

Best,
-Clint

davidlange6 commented:
ah - I missed the update!

davidlange6 commented:
+1
